lio-http-server/src/LIO/HTTP/Server/Responses.hs

```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE Trustworthy #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Taken from Simple 
+-- | This module defines some convenience functions for creating responses.
+module LIO.HTTP.Server.Responses
+  ( ok, okHtml, okJson, okXml
+  , movedTo, redirectTo
+  , badRequest, requireBasicAuth, forbidden
+  , notFound
+  , serverError
+  ) where
+
+import qualified Data.ByteString.Char8 as S8
+import qualified Data.ByteString.Lazy.Char8 as L8
+import Network.HTTP.Types
+-- import Network.Wai
+import LIO.HTTP.Server
+
+
+-- | Type alias for 'S8.ByteString'
+type ContentType = S8.ByteString
+
+-- | Creates a 200 (OK) 'Response' with the given content-type and resposne
+-- body
+ok :: ContentType -> L8.ByteString -> Response
+ok contentType body =
+  Response status200 [(hContentType, contentType)] body
+
+-- | Helper to make responses with content-type \"text/html\"
+mkHtmlResponse :: Status -> [Header] -> L8.ByteString -> Response
+mkHtmlResponse stat hdrs =
+  Response stat ((hContentType, S8.pack "text/html"):hdrs)
+
+-- | Creates a 200 (OK) 'Response' with content-type \"text/html\" and the
+-- given resposne body
+okHtml :: L8.ByteString -> Response
+okHtml body =
+  mkHtmlResponse status200 [] body
+
+-- | Creates a 200 (OK) 'Response' with content-type \"application/json\" and the
+-- given resposne body
+okJson :: L8.ByteString -> Response
+okJson = ok (S8.pack "application/json")
+
+-- | Creates a 200 (OK) 'Response' with content-type \"application/xml\" and the
+-- given resposne body
+okXml :: L8.ByteString -> Response
+okXml = ok (S8.pack "application/xml")
+
+-- | Given a URL returns a 301 (Moved Permanently) 'Response' redirecting to
+-- that URL.
+movedTo :: String -> Response
+movedTo url = mkHtmlResponse status301 [(hLocation, S8.pack url)] html
+  where html = L8.concat
+             [L8.pack
+              "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n\
+              \<HTML><HEAD>\n\
+              \<TITLE>301 Moved Permanently</TITLE>\n\
+              \</HEAD><BODY>\n\
+              \<H1>Moved Permanently</H1>\n\
+              \<P>The document has moved <A HREF=\""
+             , L8.pack url
+             , L8.pack "\">here</A>\n\
+                       \</BODY></HTML>\n"]
+
+-- | Given a URL returns a 303 (See Other) 'Response' redirecting to that URL.
+redirectTo :: S8.ByteString -> Response
+redirectTo url = mkHtmlResponse status303 [(hLocation, url)] html
+  where html = L8.concat
+             [L8.pack
+              "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n\
```

Comment:
So the doctype is for HTML 2.0, I think it's done on purpose to support as many browsers as possible. Do you still want me to switch it to HTML 5?